### PR TITLE
Remove LinkingTo: dependency from dplyr

### PR DESCRIPTION
**Pull request by krlmlr**
@lionel-: Looks like you contributed the slicing code. I suspect that the R version might run slightly slower (but not too much). We could implement a lightweight version of row subsetting in this package, but only if the performance is much worse now.

### CODE REVIEW

**Comment by lionel-** [comment]
no worries, these functions are going to be deprecated in favour of dplyr's new colwise family. I don't think the performance should be impacted much, tidyr uses a R version of this as well for nesting/unnesting IIRC.

**Comment by lionel-** [comment]
Thanks!

**Comment by krlmlr** [comment]
Thanks. When do you plan a CRAN release?

**Comment by lionel-** [comment]
@hadley would know :)

**Comment by hadley** [comment]
Not in the near future. Maybe after @lionel- has added support for progress bars and parallel computing ;)

**Comment by krlmlr** [comment]
Should be before the next dplyr release, though. Also, I'd like to get rid of `inst/include` in dplyr as soon as possible, and this means that CRAN purrr can't be installed anymore if a dev version of dplyr is installed.

**Comment by hadley** [comment]
Oh got it, we can do an earlier release if needed.

File: src/rows-data.cpp [hunk 1]
```diff
@@ -30,25 +29,19 @@ Labels::Labels(Environment execution_env_)
...
-        ++i;
-        ++j;
-      }
-      *it = j++;
-    }
+  if (!to_remove.size())
+    return;
+
+  // http://stackoverflow.com/a/22833346/946850
+  static Function subset("[.data.frame");
```
Comment (krlmlr) [inline src/rows-data.cpp]: Perhaps dplyr::slice() is better here.

**Pull request merged**